### PR TITLE
CSharp Assignments

### PR DESCRIPTION
**Pull request by SmitaPradhan2299**

### CODE REVIEW

File: Basic_CSharp/Assignment1/StringOperations.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            Console.WriteLine("After Concatination Using concat() method :");
+            Console.WriteLine(String.Concat(firstName, lastName));
+        }
+
+        //Method to compare two Strings
+        public void compareStrings()
+        {
+            Console.WriteLine("Compairing Strings using Equals() method :");
+            bool res2 = firstName.Equals(lastName);
+            if (res2 == true)
```
Comment (poojakalegithub) [inline Basic_CSharp/Assignment1/StringOperations.cs:30]: can we simplify this expression?

Comment (poojakalegithub) [reply Basic_CSharp/Assignment1/StringOperations.cs:30]: can we use string.compare?

File: Basic_CSharp/Assignment1/StringOperations.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+{
+    internal class StringOperations
+    {
+        String firstName = "Smita";
+        String lastName = "Pradhan";
+
+        //Method to concat two Strings
+        public void concatStrings()
+        {
+            Console.WriteLine("First name: " +firstName);
```
Comment (poojakalegithub) [inline Basic_CSharp/Assignment1/StringOperations.cs:17]: is there any other way to pass parameters to the console without doing {0} or +firstname?

File: Basic_CSharp/Assignment1/countriesEnum.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    internal class countriesEnum
+    {
+        public void showData()
+        {
+        Console.WriteLine("Enter Country Name:");
+        //to take input from user
+        String Country=Console.ReadLine();  
+
+        //to convert user input into upper case
+        String res=Country.ToUpper();
```
Comment (poojakalegithub) [inline Basic_CSharp/Assignment1/countriesEnum.cs]: variable name

File: Basic_CSharp/Assignment1/EmailValidation.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace PracticeCSharp.Assignments
+{
+    internal class EmailValidation
+    {
+    string email1 = "Smitanpradhan33@gmail.com";
```
Comment (poojakalegithub) [inline Basic_CSharp/Assignment1/EmailValidation.cs]: email1?

File: Basic_CSharp/Assignment1/EmailValidation.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    public void validateEmail()
+        {
+            if (email1.Contains('@') == true && email1.EndsWith(".com") == true)
```
Comment (poojakalegithub) [inline Basic_CSharp/Assignment1/EmailValidation.cs]: can we simplify this expression? Also what if the domain is different?

File: Basic_CSharp/Assignment1/Person.cs [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+namespace PracticeCSharp.Assignments
+{
+    internal class Person
+    {
+        public string FirstName { get; set; }
+
+        public string LastName { get; set; }
+
+        public byte Age { get; set; }
```
Comment (poojakalegithub) [inline Basic_CSharp/Assignment1/Person.cs:16]: why you have used byte for age?

File: Basic_CSharp/Assignment1/Employee.cs [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PracticeCSharp.Assignments
+{
+        internal class Employee : Person
+        {
+            public string JobTitle { get; set; }
```
Comment (poojakalegithub) [inline Basic_CSharp/Assignment1/Employee.cs:11]: check variable declaration standards

File: Basic_CSharp/Assignment2/Student.cs [hunk 8]
```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PracticeCSharp.Assignment2
+{
+    class Student
```
Comment (poojakalegithub) [inline Basic_CSharp/Assignment2/Student.cs:9]: what will be the default access modifier if we do specify it?

File: Basic_CSharp/Assignment2/Student.cs [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+        {
+            return ID + "  " + Name + "   " +Age + "   " + Class;
+        }
+
+        public void StudentList()
+        {
+
+            // Creating list of type Student 
+            List<Student> students = new List<Student>()
+    {
```
Comment (poojakalegithub) [inline Basic_CSharp/Assignment2/Student.cs]: indentations

File: Basic_CSharp/Assignment2/Employee.cs [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+        string Name;
+        int Salary;
+        string Department;
+
+        public override string ToString()
+        {
+            return ID + " " + Name + " " + Salary + " " + Department;
+        }
+
+        public void sortEmp()
```
Comment (poojakalegithub) [inline Basic_CSharp/Assignment2/Employee.cs]: function name

**Review by poojakalegithub — CHANGES_REQUESTED** [review]
I have reviewed assignment 1 and 2, commit the changes as per comments, then you are good to merge this PR.

File: Basic_CSharp/Assignment1/StringOperations.cs (hunk 1 above)
Comment (SmitaPradhan2299) [reply Basic_CSharp/Assignment1/StringOperations.cs:30]: Yes, we can.
I have updated the code using compare method.

File: Basic_CSharp/Assignment1/StringOperations.cs (hunk 2 above)
Comment (SmitaPradhan2299) [reply Basic_CSharp/Assignment1/StringOperations.cs:17]: Yes, we can insert a $ character before the string then an interpolation expression must contained inside an opening and closing brace, ({ and })

File: Basic_CSharp/Assignment1/countriesEnum.cs (hunk 3 above)
Comment (SmitaPradhan2299) [reply Basic_CSharp/Assignment1/countriesEnum.cs]: Updated variable name.

File: Basic_CSharp/Assignment1/EmailValidation.cs (hunk 4 above)
Comment (SmitaPradhan2299) [reply Basic_CSharp/Assignment1/EmailValidation.cs]: Updated.

File: Basic_CSharp/Assignment1/Person.cs (hunk 6 above)
Comment (SmitaPradhan2299) [reply Basic_CSharp/Assignment1/Person.cs:16]: A maximum value byte can store  is +-127 and i thought that is enough to store age.

File: Basic_CSharp/Assignment1/Employee.cs (hunk 7 above)
Comment (SmitaPradhan2299) [reply Basic_CSharp/Assignment1/Employee.cs:11]: Yes

File: Basic_CSharp/Assignment2/Student.cs (hunk 8 above)
Comment (SmitaPradhan2299) [reply Basic_CSharp/Assignment2/Student.cs:9]: Internal

File: Basic_CSharp/Assignment2/Employee.cs (hunk 10 above)
Comment (SmitaPradhan2299) [reply Basic_CSharp/Assignment2/Employee.cs]: Changed the function name.

File: Basic_CSharp/Assignment1/EmailValidation.cs (hunk 5 above)
Comment (SmitaPradhan2299) [reply Basic_CSharp/Assignment1/EmailValidation.cs]: Updated the code so that domain like  .com , .in and .org can also get validated

**Review by poojakalegithub — APPROVED** [review]

**Pull request merged**